core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -1003,8 +1003,12 @@ private static void tryWriteTempFile(Path path) throws IOException {
         if (Files.exists(path)) {
             Path resolve = path.resolve(".es_temp_file");
             try {
-                Files.createFile(resolve);
-                Files.deleteIfExists(resolve);
```

Comment:
we don't necessarily move the delete to a `finally` block as we always try to delete the left over temp file at the first.